SparkConnector/sparkconnector/configuration.py

```diff
@@ -269,10 +269,13 @@ def configure(self, opts, ports):
         conf.set('spark.kubernetes.namespace', os.environ.get('SPARK_USER'))
         conf.set('spark.master', self._retrieve_k8s_master(os.environ.get('KUBECONFIG')))
 
-        # Configure shuffle if running on K8s with Spark 3.x.x
-        if self.get_spark_version().split('.')[0]=='3':
-            conf.set('spark.shuffle.service.enabled', 'false')
-            conf.set('spark.dynamicAllocation.shuffleTracking.enabled', 'true')
+        # The image used by the Spark executor should be set by the spawner, as we need
+        # different images for different platform types
```

Comment:
Can't we check the platform directly in runtime and adapt then? (instead of having to set an extra config in the spawner, that might be out of sync)

Reply:
Which was already there.

Reply:
What is already there exactly? We make available the tag as an environment variable (VERSION_DOCKER_IMAGE), but not the whole URL. Also, note that Luca uses his own images, so not even that tag is useful. In the current solution, the URLs are hardcoded in the SparkConnector's code, which should be avoided.

@LucaCanali @diocas we could think of just reusing the SWAN image for the Spark executors, get the tag from the environment (which is already there) and the rest of the URL from some configuration file for the SparkConnector -- what is dynamic is the tag, not the URL itself, but better in a config file than in the SparkConnector's code.